From 16ea9f8f20d47665117948e2f1ad9ba48131dfea Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Mon, 3 May 2021 20:00:38 +0000
Subject: [PATCH] perfetto: Fix build with gcc-11

* fixes:
FAILED: obj/third_party/perfetto/src/trace_processor/db/lib/column.o
i686-webos-linux-g++  -m32 -march=core2 -mtune=core2 -msse3 -mfpmath=sse -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/jenkins/mjansa/build/webos-pro/honister/BUILD/work/qemux86-webos-linux/webruntime/87.0.4280.88-16-r47.2/
recipe-sysroot -MMD -MF obj/third_party/perfetto/src/trace_processor/db/lib/column.o.d -DENABLE_BROWSER_CONTROL_WEBAPI=1 -DENABLE_SAMPLE_WEBAPI=1 -DENABLE_MEMORYMANAGER_WEBAPI=1 -DENABLE_NETWORK_ERROR_PAGE_CONTROLLER_WEBAPI=1 -DOZONE_PLATFORM_WAYLAND_EXTERNAL=1 -DOS_WEBOS=1 -DNEVA_VIDEO_HO
LE=1 -DUSE_NEVA_APPRUNTIME=1 -DENABLE_WEBM_VIDEO_CODECS=1 -DUSE_CBE=1 -DUSE_PMLOG=1 -DUSE_NEVA_SUSPEND_MEDIA_CAPTURE=1 -DUSE_FILESCHEME_CODECACHE=1 -DUSE_LOCAL_STORAGE_MANAGER=1 -DWEBOS_SUBMISSION_NUMBER=16 -DUSE_LTTNG=1 -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_OZONE=1
-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -DCR_SYSROOT_HASH=d967bcef40477dbc39acef141ff22bf73f3e7cdb -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DPERFETTO_IMPLEMENTATION -I../../git/src/nev
a -Igen/neva -I../../git/src -Igen -I../../git/src/third_party/perfetto -I../../git/src/third_party/perfetto/include -Igen/third_party/perfetto/build_config -Igen/third_party/perfetto -Wno-unused-variable -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -funwind-tables -fPI
C -pipe -pthread -m32 -msse2 -mfpmath=sse -mmmx -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -Wall -Wno-unused-local-typedefs -Wno-maybe-uninitialized -Wno-deprecated-declarations -Wno-comments -Wno-packed-not-aligned -Wno-attributes -Wno-missing-field-initializers
 -Wno-unused-parameter -O2 -fno-ident -fdata-sections -ffunction-sections -fno-omit-frame-pointer -g0 -fvisibility=hidden -std=gnu++14 -Wno-narrowing -Wno-class-memaccess -fno-exceptions -fno-rtti --sysroot=../../recipe-sysroot -fvisibility-inlines-hidden -c ../../git/src/third_party/perfe
tto/src/trace_processor/db/column.cc -o obj/third_party/perfetto/src/trace_processor/db/lib/column.o
In file included from ../../git/src/third_party/perfetto/src/trace_processor/db/column.h:22,
                 from ../../git/src/third_party/perfetto/src/trace_processor/db/column.cc:17:
../../git/src/third_party/perfetto/src/trace_processor/containers/string_pool.h: In static member function 'static const uint8_t* perfetto::trace_processor::StringPool::ReadSize(const uint8_t*, uint32_t*)':
../../git/src/third_party/perfetto/src/trace_processor/containers/string_pool.h:256:34: error: 'numeric_limits' is not a member of 'std'
  256 |     PERFETTO_DCHECK(value < std::numeric_limits<uint32_t>::max());
      |                                  ^~~~~~~~~~~~~~
../../git/src/third_party/perfetto/src/trace_processor/containers/string_pool.h:256:57: error: expected primary-expression before '>' token
  256 |     PERFETTO_DCHECK(value < std::numeric_limits<uint32_t>::max());
      |                                                         ^
../../git/src/third_party/perfetto/src/trace_processor/containers/string_pool.h:256:60: error: '::max' has not been declared; did you mean 'std::max'?
  256 |     PERFETTO_DCHECK(value < std::numeric_limits<uint32_t>::max());
      |                                                            ^~~
---
 .../perfetto/src/trace_processor/containers/string_pool.h        | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/third_party/perfetto/src/trace_processor/containers/string_pool.h b/src/third_party/perfetto/src/trace_processor/containers/string_pool.h
index 11ae91cfec..558ff00e8f 100644
--- a/src/third_party/perfetto/src/trace_processor/containers/string_pool.h
+++ b/src/third_party/perfetto/src/trace_processor/containers/string_pool.h
@@ -20,6 +20,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <limits>
 #include <unordered_map>
 #include <vector>
 
